### PR TITLE
feat: Convert nrg-application to glimmer component

### PR DESCRIPTION
**Pull request by KeithClinard**

### CODE REVIEW

File: addon/components/nrg-application.js [hunk 1]
```diff
@@ -1,59 +1,70 @@
...
+    this.resize.off('didResize', this, this.setMainContentStyle);
+  }
+
+  @action
+  toggleSidebar() {
+    if (this.responsive.isComputerScreenGroup) {
+      this.application.sidebarMenuIsOpen = !this.application.sidebarMenuIsOpen;
+    } else {
+      this.sidebarIsOpen = !this.sidebarIsOpen;
+    }
```
Comment (TSenter) [inline addon/components/nrg-application.js:63]:
```suggestion
    let property = 'sidebarIsOpen';
    if (this.responsive.isComputerScreenGroup) {
      property = 'application.sidebarMenuIsOpen';
    }
    this.toggleProperty(property);
```

File: addon/components/nrg-application.hbs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{{! template-lint-disable no-inline-styles }}
+<div
+  class="nrg-application
+    {{if this.fullscreenMap "fullscreen-map"}}
+    {{if this.computerScreenSidebarActive "large-screen-sidebar-active"}}"
+  {{did-insert this.onInsert}}
+  {{will-destroy this.onDestroy}}
+>
+  <NrgAppbar @title={{this.title}} @toggleSidebar={{this.toggleSidebar}} />
```
Comment (TSenter) [inline addon/components/nrg-application.hbs:9]:
```suggestion
  <NrgAppbar @title={{@title}} @toggleSidebar={{this.toggleSidebar}} />
```

File: addon/components/nrg-application.hbs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  <div class="pushable">
+    <NrgSidebar
+      @isOpen={{this.sidebarIsOpen}}
+      @clickedSidebarItem={{this.clickedLink}}
+    />
+    {{#if this.fullscreenMap}}
```
Comment (TSenter) [inline addon/components/nrg-application.hbs:15]:
```suggestion
    {{#if @fullscreenMap}}
```

File: addon/components/nrg-application.js [hunk 4]
```diff
@@ -1,59 +1,70 @@
...
+    );
+  }
+
+  constructor() {
+    super(...arguments);
+    this.setMainContentStyle();
+  }
+
+  setMainContentStyle() {
+    return htmlSafe(`height:${window.innerHeight}px`);
```
Comment (TSenter) [inline addon/components/nrg-application.js:44]:
```suggestion
  setMainContentStyle() {
    this.mainContentStyle = htmlSafe(`height:${window.innerHeight}px`);
```

File: addon/components/nrg-application.hbs [hunk 5]
```diff
@@ -0,0 +1,59 @@
+{{! template-lint-disable no-inline-styles }}
+<div
+  class="nrg-application
+    {{if this.fullscreenMap "fullscreen-map"}}
```
Comment (TSenter) [inline addon/components/nrg-application.hbs:4]:
```suggestion
    {{if @fullscreenMap "fullscreen-map"}}
```

**Review by TSenter — CHANGES_REQUESTED** [review]

File: addon/components/nrg-application.js (hunk 1 above)
Comment (KeithClinard) [reply addon/components/nrg-application.js:63]: `toggleProperty` does not exist in octane components:
image

Comment (TSenter) [reply addon/components/nrg-application.js:63]: Ah my bad, I read the docs for the Ember component that still has them, not the Glimmer component.

**Review by TSenter — APPROVED** [review]

**Pull request merged**